backend/src/index.ts

```diff
@@ -0,0 +1,151 @@
+import express, { Request, Response } from 'express';
+import cors from 'cors';
+import { prisma } from './database'
+import { errorResponseBuilder, generateRandomPassword, isValidUser, parseUserForResponse } from './utils';
+
+const app = express();
+app.use(express.json());
+app.use(cors())
+
+// Create new user
+app.post('/users/new', async (req: Request, res: Response) => {
+  const errorBuilder = errorResponseBuilder(res);
+
+  try {
+    const userData = req.body;
+
+    // Validate input
+    if(!isValidUser(req.body)){
+      return errorBuilder.validationError()
+    }
+    
+    // Check if the user exists or not
+    const existingUserByEmail = await prisma.user.findFirst({ where: { email: req.body.email }});
+    if (existingUserByEmail) {
+      return errorBuilder.emailAlreadyInUse()
+    }
+    
+    const existingUserByUsername = await prisma.user.findFirst({ where: { username: req.body.username }});
+    if (existingUserByUsername) {
+      return errorBuilder.usernameAlreadyTaken()
+    }
+
+    // Success case
+    const user = await prisma.user.create({ 
+      data: { ...userData, password: generateRandomPassword(10) } 
+    });
+    
+    return res.status(201).json({       
+      error: undefined, 
+      data: parseUserForResponse(user), 
+      succes: true 
+    });
+  } catch (error) {
+    console.log(error)
+    // Return a failure error response
+    return errorBuilder.serverError()
+  }
+});
+
+// Edit a user
+app.post('/users/edit/:userId', async (req: Request, res: Response) => {
+
+  const errorBuilder = errorResponseBuilder(res);
+
+  try {
+
+    const {id,password, ...userData} = req.body;
+
```

Comment:
```suggestion
    const {id,password, ...userData} = req.body;
```
Maybe use eslint or something to narrow down the spaces between lines of code

Reply:
Good idea, I was too lazy to set up formatting. I'll set up prettier + some pre-commit hooks to take care of formatting.